subsys/bluetooth/host/gatt.c

```diff
@@ -813,7 +886,28 @@ static void sc_indicate(uint16_t start, uint16_t end);
 static void db_hash_process(struct k_work *work)
 {
 #if defined(CONFIG_BT_SETTINGS)
```

Comment:
Maybe we could perform `db_hash_process` in `db_hash_commit` callback context (instead of moving it to Bluetooth long workqueue context that has low priority)?
That could allow us to avoid synchronization issues related to processing the db_hash and marking peers as change unaware when GATT database changes after a DFU operation. I think current implementation introduces possibility of accessing GATT by previously bonded peer before DB hash is processed and before the peer is properly marked as change unaware.
The issue also seem not directly related to the PR so may be addressed separately I think.

Reply:
Yeah, please open another low-prio bug for that. you can assign me for now as well.
I think it should be okay to NAK all GATT access until the hash has been calculated.

Reply:
Alternatively, maybe we could delay sending GATT response and force the database hash calculation in the current context (I am not sure which solution would provide better interoperability with devices available on the market)

Reply:
that may be a valid idea, but this is kinda why we added this long-wq in the first place: bt runs as a coop prio, and the hash can be quite long, and will pre-empt other threads from running (even if cooperative).

Reply:
Follow-up issue: https://github.com/zephyrproject-rtos/zephyr/issues/54773